### ISSUE TITLE
Replacing the animation with another one doesn't work

### ISSUE THREAD

**Issue by appsandkids** [pisi/Reel#162]
Hi!

I am trying to use the plugin in the following way: play animations in random sequence. Eg. animation1, animation2, animation5, animation3, ...

I am checking the frames in "frameChange" function and when "frame == numberOfFrames" I'm kicking off the new animation which basically calls $(imgElementID).reel( ...options... ) again.

The problem: according to documentation, this should trigger the "teardown" first and delete the current instance. And it does - teardown is triggered. However the plugin doesn't wait for that to finish! I think it just continues to setup the new animation with the old still there. And it just doesn't run.

Although it seems that you have foreseen this option (since you call teardown and write about it), however something is not working.

**Comment by appsandkids** [comment]
By the way, just wanted to add that it works if I trigger "teardown" myself and call .reel after a timeout of 2 seconds. And 1 second was not enough (at least testing on mobile with phonegap).

Therefore it would be best if:
1. either we can get some notification/event triggered when "teardown" is complete
   or
2. the plugin takes care of it by itself and doesn't proceed with .reel() actions while "teardown" still in progress.

By the way, thank you for a great plugin!

**Comment by pisi** [comment]
Hello,

I reviewed all the pieces involved in calling `.reel()` again. When [`.reel()` method](http://test.vostrel.cz/jquery.reel/docs/jquery.reel.html#reel-Method) is called, first, the given collection is filtered and only `<img>` tags with specified width and height are allowed to stay in the collection. Second, this resulting collection is acted upon, which makes Reel start. Now, during the filtering, targets being Reels already have `.unreel()` applied to each of them and only AFTER the teardown of all is completely finished, only then the resulting collection is formed and passed to setup procedure. The [`teardown` event](http://test.vostrel.cz/jquery.reel/docs/jquery.reel.html#teardown-Event) itself doesn't do anything asynchronous.

I realize this may be too out-dated now (so sorry), but can you please send me your code for inspection? You may not be doing everything right.

**Comment by appsandkids** [comment]
Hi Petr,

thank you for replying. I will try to find the code (probably the next
week).

Eventually I have chosen a native "animateImages" solution. I have quite
big images in the sequence and I need quick reaction times (i.e. "reaction"
animation on touches). Unfortunately I could not make pisi reel (and I
tried jsMovie too) to run fast enough on the iphone (and even on the iPad).

However if you say I might have done something wrong, I will be glad to
show you what I had. Perhaps the performance issues are also related to
this.

Best wishes,
Ole

**Comment by pisi** [comment]
Hi Ole,

sorry to hear, I struggle for great performance on these devices and tests run smooth even on a lo-fi Android 2.3.5. However as Reel has many additional interaction-related processing to do, it can lose its power under some specific conditions including image dimensions, number of frames, sprite or sequence, format of images and compression (if applicable), animation speed and last but not least, the tempo. However you can adjust how Reel treats iPhones, iPods, Androids and Blackberries.

Reel, in order to ease things on these _lazy_ devices, adjusts the clock ticking. The [`tempo`](http://jquery.vostrel.cz/reel#tempo) of its internal ticker gets divided by [`laziness`](http://jquery.vostrel.cz/reel#laziness) factor, which by default slows down the tick frequency to 1/6 of the full ticking speed. Default `tempo: 36` and `laziness: 6` results in (`36 / 6 = 6`) 6 ticks per second. This much longer tick allows for the processing of all the additional interaction events amplified by a slow CPU.

In short, keep your image sizes sane and try to play with the `laziness` setting - higher values will result in longer ticks allowing for longer processing in-between ticks.

**Comment by pisi** [comment]
FYI, there is a new method for image switching being developed in issue #190, which allows seamless replacing of image(s).

**Comment by pisi** [comment]
As much I want to have issue resolutions verified, I want such long resolved ones out of the way. Verification still welcome tho!
